### ISSUE TITLE
Partial logConcurrentOutput

### ISSUE THREAD

**Issue by aicioara** [sindresorhus/grunt-concurrent#68]
At the moment there just a switch for logging: either log all at the end or non-deterministically intersect the logs. How about being able to specify one file for continuous logging and rest for bulk logging. This way you get continuous feedback while also keeping a good history of the log.

A scenario for this would be: User has 2 tasks which each run for 5 minutes and both have massive output. We want continuous feedback from task1 to figure out the ETA and then bulk the output of the other task at the very end.

I am happy to try open a pull request if any of the choices are part of the philosophy of the library.

**Comment by sindresorhus** [comment]
Can you show an example of how this would look like (in the gruntfile)? I'm not completely sure what you're looking for exactly.

**Comment by aicioara** [comment]
Do you mean, how would the configuration point look like? 

If this was a new feature, I would expect to write in the configuration file something like this 
```javascript
var concurrentConfig = {
  tasks: ['task1', 'task2', 'task3'],
  options: {
    logWhileRunning: 'task1'
  }
}
```

The above would continuously print whatever task1 output is, and then when all tasks are done, it will output the output of task2 and task3 in any order (or ideally the order specified in the tasks array)

**Comment by sindresorhus** [comment]
I'm thinking renaming `logConcurrentOutput` to `streamingLog` (name suggestions welcome) (while still having an internal deprecated alias to `logConcurrentOutput` internally). In addition to supporting the old boolean value it could also accept an array of tasks to log concurrently, like you want. We then have one option that can handle all use-cases. Thoughts?

**Comment by aicioara** [comment]
how would the new `streamingLog` work? Is it just a straight rename?

**Comment by sindresorhus** [comment]
Yes, just IMHO a clearer name.

**Comment by aicioara** [comment]
How about the following super useful scenario

Say we have task1, task2 and task3. Timewise, 
- Task 1 - 1:00
- Task 2 - 3:00
- Task 3 - 2:00

We use an API point:
```javascript
logging: "parallel" | "stored" | "smart"
// open to option suggestion
```

the way it will work
- Option `parallel` works just like having the flag `logConcurrentOutput` set to true
- Option `stored` works just like having the flag `logConcurrentOutput` set to false
- Option `smart` continuously logs task 1 for the first minute, then dumps the first minute of task 2 and continues continuously logging task 2, and then dumps the output of task 3 (since task 3 is already finished)

**Comment by aicioara** [comment]
This way with option "smart" (again let's rename it to something less cheesy) you have continuous feedback while your logs are logically correct.

**Comment by sindresorhus** [comment]
I don't think there's any way to make the `smart` mode smart enough not to be annoying. 1 minute is just arbitrary. No matter what we pick it will be wrong for some. I'd rather have something explicit.

**Comment by aicioara** [comment]
No, no, that was just my particular example so it is easier to understand. Let me try again.

- Task 1 - X min
- Task 2 - Y min
- Task 3 - Z min

Option smart 
- continuously logs task 1 for X minutes, 
- then dumps the first X minutes of task 2 
- then continues logging task 2 for Math.max(Y - X, 0) minutes
- then dumps the first Math.max(X, Y) minutes output of task 3 
- then continues logging task 3 for Math.max(Z - Math.max(X, Y), 0)

Does it make sense?

**Comment by aicioara** [comment]
Also, you don't need to know the number of minutes of each task. It is for illustration only. It will work like this. 

- Log output of task1 until it is done. In the time cache the output of task 2 and 3.
- Then when task 1 is done, dump whatever was cached for task2 so far and carry on logging output of task2
- etc

**Comment by sindresorhus** [comment]
Sorry, forgot about this one. I can see the usefulness, but I don't think I want to maintain such complexity. Happy to link to a fork that does it.

**Comment by xizhao** [comment]
This would be a huge +1 for me.  Any reconsideration to reopen?

**Comment by aicioara** [comment]
I could submit a patch if this is of interest. It should not be such a big architecture change in my opinion.
